services-custom/s3-transfer-manager/README.md

```diff
@@ -21,88 +22,131 @@ First, you need to include the dependency in your project.
+                                                       .sourceBucket("source_bucket")
+                                                       .sourceKey("source_key")
+                                                       .destinationBucket("dest_bucket")
+                                                       .destinationKey("dest_key")
+                                                       .build();
+CopyRequest copyRequest = CopyRequest.builder()
+                                     .copyObjectRequest(copyObjectRequest)
+                                     .build();
+
+Copy copy = transferManager.copy(copyRequest);
+
+// Wait for the transfer to complete
+CompletedCopy completedCopy = copy.completionFuture().join();
 ```
 
-Refer to the static factory methods available in `AsyncResponseTransformer` for other destinations.
+### Transfer multiple objects in the same directory
 
-### Attach a TransferListener
+#### Upload a local directory to an S3 bucket
 
-To monitor a transfer's progress, you can include a `TransferListener` with your transfer request:
+To upload a local directory recursively to an S3 bucket, you need to provide the source directory and the target bucket.
 
 ```java
-FileUpload upload =
-    tm.uploadFile(u -> u.source(Paths.get("myFile.txt"))
-                        .putObjectRequest(p -> p.bucket("bucket").key("key"))
-                        .addTransferListener(LoggingTransferListener.create()));
-upload.completionFuture().join();
+S3TransferManager transferManager = S3TransferManager.create();
+DirectoryUpload directoryUpload = transferManager.uploadDirectory(UploadDirectoryRequest.builder()
+                                                 .sourceDirectory(Paths.get("source/directory"))
+                                                 .bucket("bucket")
+                                                 .build());
+
+// Wait for the transfer to complete
+CompletedDirectoryUpload completedDirectoryUpload = directoryUpload.completionFuture().join();
+
+// Print out the failed uploads
+completedDirectoryUpload.failedTransfers().forEach(System.out::println);
 ```
 
-You can provide your own implementation of a `TransferListener` to implement progress-bar-type functionality.
\ No newline at end of file
+#### Download S3 objects within the same bucket to a local directory
+
+To download all S3 objects within the same bucket, you need to provide the destination directory and the source bucket.
+
+```java
+S3TransferManager transferManager = S3TransferManager.create();
+         DirectoryDownload directoryDownload =
+             transferManager.downloadDirectory(
+                  DownloadDirectoryRequest.builder()
+                                          .destination(Paths.get("destination/directory"))
+                                          .bucket("bucket")
+                                          .listObjectsV2RequestTransformer(l -> l.prefix("prefix"))
```

Comment:
there's no explanation for what this line does (`listObjectsV2RequestTransformer`)

Reply:
I think we should probably try to keep the sample code simple. We have more docs explaining this in https://github.com/aws/aws-sdk-java-v2/blob/ecb1654466244173311d6dc90dfb2e86f8d7586f/services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/model/DownloadDirectoryRequest.java#L234-L264

Reply:
On second thought, it may be better to keep it here. I'll add a comment there